Broken CMake config file for pubsub

The CMake config file generated and installed along the Pub/Sub library is broken resulting in a CMake error when trying to `find_package` it.

I believe the issue is here:
https://github.com/googleapis/google-cloud-cpp/blob/d915c7f57fa407cf58e6c78ad5401657345da990/google/cloud/pubsub/config-version.cmake.in#L19
as the `GOOGLE_CLOUD_CPP_PUBSUB_CLIENT_VERSION` variable is not defined in the CMakeLists.txt generating the config file.

**To Reproduce**
Steps to reproduce the behavior:
1. Build and intall pubsub client library
2. Attempt a `find_package(pubsub_client REQUIRED)` from a consumer CMake project
3. Error:
```
CMake Error at /usr/lib/x86_64-linux-gnu/cmake/pubsub_client/pubsub_client-config-version.cmake:19 (if):
  if given arguments:

    "EQUAL" "0"

  Unknown arguments specified
```

**Context**
* Linux Debian testing/bullseye
* google-cloud-cpp d915c7f57fa407cf58e6c78ad5401657345da990
* CMake 3.17.1

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,6 +23,9 @@ project(google-cloud-cpp CXX C)
 set(GOOGLE_CLOUD_CPP_VERSION_MAJOR 1)
 set(GOOGLE_CLOUD_CPP_VERSION_MINOR 14)
 set(GOOGLE_CLOUD_CPP_VERSION_PATCH 0)
+string(CONCAT GOOGLE_CLOUD_CPP_VERSION "${GOOGLE_CLOUD_CPP_VERSION_MAJOR}"
+              ".${GOOGLE_CLOUD_CPP_VERSION_MINOR}"
+              ".${GOOGLE_CLOUD_CPP_VERSION_PATCH}")
 
 # Configure the Compiler options, we use C++11 features by default.
 set(GOOGLE_CLOUD_CPP_CXX_STANDARD

--- a/external/googleapis/CMakeLists.txt
+++ b/external/googleapis/CMakeLists.txt
@@ -16,11 +16,6 @@
 
 cmake_minimum_required(VERSION 3.5)
 
-string(
-    CONCAT GOOGLE_APIS_CPP_PROTOS_VERSION "${GOOGLE_CLOUD_CPP_VERSION_MAJOR}"
-           "." "${GOOGLE_CLOUD_CPP_VERSION_MINOR}" "."
-           "${GOOGLE_CLOUD_CPP_VERSION_PATCH}")
-
 # Give application developers a hook to configure the version and hash
 # downloaded from GitHub.
 set(GOOGLE_CLOUD_CPP_GOOGLEAPIS_URL
@@ -170,7 +165,7 @@ function (googleapis_cpp_set_version_and_alias short_name)
     add_dependencies("googleapis_cpp_${short_name}" googleapis_download)
     set_target_properties(
         "googleapis_cpp_${short_name}"
-        PROPERTIES VERSION "${GOOGLE_APIS_CPP_PROTOS_VERSION}"
+        PROPERTIES VERSION "${GOOGLE_CLOUD_CPP_VERSION}"
                    SOVERSION ${GOOGLE_CLOUD_CPP_VERSION_MAJOR})
     add_library("googleapis-c++::${short_name}" ALIAS
                 "googleapis_cpp_${short_name}")

--- a/google/cloud/CMakeLists.txt
+++ b/google/cloud/CMakeLists.txt
@@ -19,10 +19,6 @@ set(FPHSA_NAME_MISMATCHED Threads) # Quiet warning caused by Abseil
 find_package(absl CONFIG REQUIRED)
 unset(FPHSA_NAME_MISMATCHED)
 
-string(CONCAT GOOGLE_CLOUD_CPP_VERSION "${GOOGLE_CLOUD_CPP_VERSION_MAJOR}" "."
-              "${GOOGLE_CLOUD_CPP_VERSION_MINOR}" "."
-              "${GOOGLE_CLOUD_CPP_VERSION_PATCH}")
-
 # Generate the version information from the CMake values.
 configure_file(internal/version_info.h.in
                ${CMAKE_CURRENT_SOURCE_DIR}/internal/version_info.h)

--- a/google/cloud/pubsub/config-version.cmake.in
+++ b/google/cloud/pubsub/config-version.cmake.in
@@ -16,7 +16,7 @@ set(PACKAGE_VERSION @DOXYGEN_PROJECT_NUMBER@)
 
 # This package has not reached 1.0, there are no compatibility guarantees
 # before then.
-if (@GOOGLE_CLOUD_CPP_PUBSUB_CLIENT_VERSION@ EQUAL 0)
+if (@GOOGLE_CLOUD_CPP_CLIENT_VERSION@ EQUAL 0)
     if ("${PACKAGE_FIND_VERSION}" STREQUAL "")
         set(PACKAGE_VERSION_COMPATIBLE TRUE)
     elseif ("${PACKAGE_VERSION}" VERSION_EQUAL "${PACKAGE_FIND_VERSION}")

